rayon-core/src/scope/mod.rs

```diff
@@ -262,22 +291,109 @@ where
     R: Send,
 {
     in_worker(|owner_thread, _| {
-        unsafe {
-            let scope: Scope<'scope> = Scope {
-                owner_thread_index: owner_thread.index(),
-                registry: owner_thread.registry().clone(),
-                panic: AtomicPtr::new(ptr::null_mut()),
-                job_completed_latch: CountLatch::new(),
-                marker: PhantomData,
-            };
-            let result = scope.execute_job_closure(op);
-            scope.steal_till_jobs_complete(owner_thread);
-            result.unwrap() // only None if `op` panicked, and that would have been propagated
-        }
+        let scope = Scope::<'scope>::new(owner_thread);
+        unsafe { scope.base.complete(owner_thread, || op(&scope)) }
+    })
+}
+
+/// Create a "fork-join" scope `s` with FIFO order, and invokes the
+/// closure with a reference to `s`. This closure can then spawn
+/// asynchronous tasks into `s`. Those tasks may run asynchronously with
+/// respect to the closure; they may themselves spawn additional tasks
+/// into `s`. When the closure returns, it will block until all tasks
+/// that have been spawned into `s` complete.
+///
+/// # Task execution
+///
+/// Tasks in a `scope_fifo()` run similarly to [`scope()`], but there's a
+/// difference in the order of execution. Consider a similar example:
+///
+/// [`scope()`]: fn.scope.html
+///
+/// ```rust
+/// # use rayon_core as rayon;
+/// // point start
+/// rayon::scope_fifo(|s| {
+///     s.spawn_fifo(|s| { // task s.1
+///         s.spawn_fifo(|s| { // task s.1.1
+///             rayon::scope_fifo(|t| {
+///                 t.spawn_fifo(|_| ()); // task t.1
+///                 t.spawn_fifo(|_| ()); // task t.2
+///             });
+///         });
+///     });
+///     s.spawn_fifo(|s| { // task 2
```

Comment:
task s.2?

Reply:
Yes, I changed the comments in both places to "task s.2".